### PR TITLE
unique_ptr: PBMETAD

### PR DESCRIPTION
**Pull request by GiovanniBussi**
@carlocamilloni @maxbonomi 

I replaced pointers with unique_ptr's in PBMetaD. Could you please check if what I did is ok? In case so I will merge it to master. Thanks!

Giovanni

### CODE REVIEW

**Comment by carlocamilloni** [comment]
It looks fine, but I am not very confident about my understanding of unique_ptr yet.

C

**Comment by carlocamilloni** [comment]
@giovannibussi I don't understand why in the destructor we don't need to close the files anymore

**Comment by GiovanniBussi** [comment]
@carlocamilloni I checked the code for [FileBase::~FileBase()](https://github.com/plumed/plumed2/blob/566d287a0ca50f9ed711f0a35a353ffb2c675619/src/tools/FileBase.cpp#L121) and [FileBase::close()](https://github.com/plumed/plumed2/blob/566d287a0ca50f9ed711f0a35a353ffb2c675619/src/tools/FileBase.cpp#L95) and it looks like the destructor is also closing the file. The only difference is that close() also asserts that file has been explicitly opened (not cloned from another file) and resets the local pointers to NULL (in order for the file object to be reusable).

So, in short, closing the file should not be necessary if you are going to destroy the file object.

**Comment by carlocamilloni** [comment]
ok then is good to me!
